lib/ClangImporter/ClangImporter.cpp

```diff
@@ -8275,3 +8277,34 @@ bool importer::isCxxConstReferenceType(const clang::Type *type) {
   auto pointeeType = getCxxReferencePointeeTypeOrNone(type);
   return pointeeType && pointeeType->isConstQualified();
 }
+
+AccessLevel importer::convertClangAccess(clang::AccessSpecifier access) {
+  switch (access) {
+  case clang::AS_public:
+    // C++ 'public' is actually closer to Swift 'open' than Swift 'public',
+    // since C++ 'public' does not prevent users from subclassing a type or
+    // overriding a method. However, subclassing and overriding are currently
+    // unsupported across the interop boundary, so we conservatively map C++
+    // 'public' to Swift 'public' in case there are other C++ subtleties that
+    // are being missed at this time (e.g., C++ 'final' vs Swift 'final').
+    return AccessLevel::Public;
+
+  case clang::AS_protected:
+    // Swift does not have a notion of protected fields, so map C++ 'protected'
+    // to Swift 'private'.
+    return AccessLevel::Private;
+
+  case clang::AS_private:
+    // N.B. Swift 'private' is more restrictive than C++ 'private' because it
+    // also cares about what source file the member is accessed.
+    return AccessLevel::Private;
+
+  case clang::AS_none:
+    // The fictional 'none' specifier is given to top-level C++ declarations,
+    // for which C++ lacks the syntax to give an access specifier. (It may also
+    // be used in other cases I'm not aware of.) Those declarations are globally
```

Comment:
This is a bit complicated. It is not necessarily true that top-level C++ declarations are globally visible. 
A couple of ways they could have different visibility:
* They could be in an anonymous namespace
* The could have `__attribute__((visibility(...)))` annotations on them

I think it is unlikely we would encounter the former in a module/header file, but we might still run into the latter. If that is the case, I wonder if that could trigger linker errors instead of some compilation diagnostics. That being said, I am not sure if this code path needs to know about any of that, I just wanted to point out there is more to the visibility of top level declarations.

Reply:
In those cases, would Clang report something other than `clang::AS_none`?

(I will test this)

Reply:
So, it looks like `__attribute__((visibility(...)))` is not referring to access specifiers, but visibility of the symbol during linkage: [GCC docs for visibility attribute](https://gcc.gnu.org/onlinedocs/gcc/Common-Function-Attributes.html#index-visibility-function-attribute)

In other words, I think it's orthogonal to what I mean by "globally visible" here, by which I mean "globally visible within the compilation unit, at the source level."

Reply:
I believe that is also the case for anonymous namespaces, whose purpose is to hide them from the object file interface, rather than from source files including the header file.

Reply:
> "globally visible" here, by which I mean "globally visible within the compilation unit, at the source level."

We are importing C++ constructs from a different module that we will be linked against. So I am not sure if this is completely orthogonal. Referring to those symbols might result in linker errors, so maybe we should not even attempt to import symbols like that.

That being said, you are right that this discussion is out of scope of this PR, something that we could look into independently at some point.